src/jdk.jfr/share/classes/jdk/jfr/internal/instrument/FileChannelImplInstrumentor.java

```diff
@@ -114,22 +132,30 @@ public int read(ByteBuffer dst, long position) throws IOException {
     @JIInstrumentationMethod
     public long read(ByteBuffer[] dsts, int offset, int length) throws IOException {
         FileReadEvent event = FileReadEvent.EVENT.get();
+        FileReadIOStatisticsEvent readPeriodicEvent = FileReadIOStatisticsEvent.EVENT.get();
+        long bytesRead = 0;
+
+        long startTime = System.nanoTime();
         if (!event.isEnabled()) {
             return read(dsts, offset, length);
```

Comment:
if the FileReadEvent is not enabled, then this will return and not collect the aggregate data, I think this needs to match the other streams

Reply:
Thanks for pointing it out. I've now updated the code to capture read bytes/write bytes even when the 'fileReadEvent'/'fileWriteEvent' is disabled.